src/core/net/ip6.cpp

```diff
@@ -1037,6 +1037,19 @@ Error Ip6::PassToHost(OwnedPtr<Message> &aMessagePtr,
     }
 #endif
 
+#if defined(__ANDROID__)
```

Comment:
The CastOS also defines the `__ANDROID__`. Does this change also applies to CastOS?

Reply:
Thanks for pointing out! This does not apply to CastOS, added a new macro `OPENTHREAD_CONFIG_ANDROID_ENABLE` for it.

Reply:
can we reuse `OPENTHREAD_CONFIG_ANDROID_NDK_ENABLE`?

Reply:
Looks like `OPENTHREAD_CONFIG_ANDROID_NDK_ENABLE` shouldn't be reused here @bukepo 
This flag is for `building ot-cli, ot-daemon, and ot-ctl using the Android NDK CMake toolchain file` [#9010], it won't be enabled on aosp ([#9776] for reference).

Reply:
Updated the flag name to `OPENTHREAD_CONFIG_ANDROID_AOSP_ENABLE` to be more specific.